windows/test-app.mjs

```diff
@@ -0,0 +1,228 @@
+
+export function linkSourceFiles(
+    srcPath,
+    destPath,
+    relativeDestDir
+) {
+    return Promise.all(walk(srcPath).map((absoluteSrcFilePath) => {
+        if (cppSourceFilesExtensions.indexOf(path.extname(absoluteSrcFilePath)) != -1) {
+            const filename = path.relative(srcPath, absoluteSrcFilePath);
+            const fullDestPath = path.join(destPath, relativeDestDir, filename);
+
+            //Avoid creating links when they already exist
+            if (!fs.existsSync(fullDestPath)) {
+                fs.link(absoluteSrcFilePath, fullDestPath, err => {
+                    if (err) {
+                        throw err;
+                    }
+                });
+            }
+        }
+    }));
+}
+
+export async function copyProjectTemplateAndReplace(srcRootPath, destPath, reactNativeModulePath, testAppModulePath) {
+    if (!srcRootPath) {
+        throw new Error('Need a path to copy from');
+    }
+
+    if (!destPath) {
+        throw new Error('Need a path to copy to');
+    }
+
+    if (!reactNativeModulePath) {
+        throw new Error('react-native-windows node module is not installed');
+    }
+
+    if (!testAppNodeModulePath) {
+        throw new Error('react-native-test-app node module is not installed');
+    }
+
+    const projDir = 'ReactTestApp';
+
+    fs.mkdirSync(path.join(destPath, windowsDir, projDir, 'Bundle'), { 'recursive': true });
+
+    const manifestFilePath = findClosestPathTo('app.json');
+
+    //Read path to resources from manifest and copy them
+    fs.readFile(manifestFilePath, function (error, content) {
+        var resourcesPaths = JSON.parse(content.toString()).resources;
+        if (!resourcesPaths) {
+            throw new Error('No resources specified in app.json');
+        }
+        resourcesPaths = resourcesPaths.windows ? resourcesPaths.windows : resourcesPaths;
+        for (const resource of resourcesPaths) {
+            copyAndReplaceAll(path.relative(path.dirname(manifestFilePath), resource), destPath, path.join(windowsDir, projDir, 'Bundle', path.basename(resource)));
+        }
+    });
+
+    const replacements = {
+        '\\$\\(ManifestRootPath\\)': path.relative(path.join(destPath, windowsDir, projDir), path.dirname(manifestFilePath)),
+        '\\$\\(ReactNativeModulePath\\)': path.relative(path.join(destPath, windowsDir, projDir), reactNativeModulePath),
+        '\\$\\(SourceFilesPath\\)': path.relative(path.join(destPath, windowsDir, projDir), path.join(srcRootPath, projDir)),
+    };
+
+    const projectFilesMappings = [
+        'ReactTestApp.vcxproj',
+        'PropertySheet.props',
+        'Package.appxmanifest',
+        'ReactTestApp.vcxproj.filters',
+        'ReactTestApp_TemporaryKey.pfx',
+        'packages.config'
+    ].map((file) => ({
+        from: path.join(srcRootPath, projDir, file),
+        to: path.join(windowsDir, projDir, file)
+    }));
+
+    for (const mapping of projectFilesMappings) {
+        await copyAndReplace(
```

Comment:
Do we have to wait for the copy/replace to complete? If not, we can remove `await`.